### ISSUE TITLE
Help wanted, to understned results

### ISSUE THREAD

**Issue by oduled** [RobTillaart/INA228#15]
Hello Rob,
I run INA228_performance.ino on INA228 Current Voltage Monitoring Sensor Module that I bought from - https://www.aliexpress.com/item/1005007670304728.html?spm=a2g0o.order_detail.order_detail_item.3.21adf19cjncOVI.
with esp32 , Please see the result that I got. What I can understand from it?
[INA228log.txt](https://github.com/user-attachments/files/18135541/INA228log.txt)

**Comment by RobTillaart** [comment]
I'm ill and not able to work, might take some weeks.

**Comment by oduled** [comment]
Sorry to hear, Get well soon

**Comment by RobTillaart** [comment]
Still recovering but found some time to think about the problem.
Does it still exist?

(relevant snippet of the log file)
```
C:\INA228_performance.ino

INA228_LIB_VERSION: 0.1.3


========================================================
Speed: 100000

test_core
getBusVoltage:	674
getShuntVoltage:	1252
getCurrent:	673
getPower:	673
getTemperature:	580
getEnergy:	866
getCharge:	857

========================================================
Speed: 200000

test_core
getBusVoltage:	380
getShuntVoltage:	713
getCurrent:	380
getPower:	381
getTemperature:	333
getEnergy:	476
getCharge:	477

========================================================
Speed: 400000

test_core
getBusVoltage:	248
getShuntVoltage:	473
getCurrent:	248
getPower:	249
getTemperature:	226
getEnergy:	299
getCharge:	300

========================================================
Speed: 800000

test_core
getBusVoltage:	194
getShuntVoltage:	381
getCurrent:	194
getPower:	195
getTemperature:	187
getEnergy:	223
getCharge:	223

Doneets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:4832
load:0x40078000,len:16460
load:0x40080400,len:4
load:0x40080404,len:3504
entry 0x400805cc
```

Apparently the sketch crashes after the last performance test. 
It might be that 
1. the processor has problems with the last used I2C speed of 800 000. 
2. The I2C pins are using reserved pins?


Could you adapt the code of the so it only tests up to 400 000 ?
Which pins are used for I2C?

**Comment by RobTillaart** [comment]
@oduled 

> rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)

Googled on that error message and several sites point to a power issue, e.g. 
- https://github.com/espressif/arduino-esp32/issues/4404

Can you verify your power sources?

**Comment by RobTillaart** [comment]
@oduled

Any new insights to report?

**Comment by oduled** [comment]
Sorry about the delay response, 
I checked it with "INA228_ demo" example and it works OK, also when I change CLK frequency to 100K, 800K and also 1M.

**Comment by RobTillaart** [comment]
Good to hear it works, so the clock seems not to cause the RESET. 

Do both sketches use the same pins?

**Comment by oduled** [comment]
Yes, Use esp32 GPIO21 & GPIO22, example "INA228_ demo" work OK but INA228_performance is getting different result with different CLK

**Comment by RobTillaart** [comment]
> INA228_performance is getting different result with different CLK

Can you post the output and explain what you got versus what you expected?

**Comment by oduled** [comment]
INA228log.txt
Please see result that I attached when i open that issue.
Result show that when changing clk, result was changed

**Comment by RobTillaart** [comment]
OK, 
The performance sketch returns the duration in microseconds of the commands, not the actual values.

code snippet 
```cpp
  start = micros();
  f = INA.getBusVoltage();
  stop = micros();
  Serial.print("getBusVoltage:\t");
  Serial.print(stop - start);    <<<<<<<<<<<<<<<<  does not print the value  INA.getBusVoltage()
  Serial.println();
  delay(10);
```

So what you see is that higher I2C clocks speeds up the execution time of the code.
- note it does nto scale linearly
- 400 kHz is about 2.5 x faster than 100 kHz
- 800 kHz is about 3.2 x faster than 100 kHz (but definitely more error prone, so only interesting if every us counts.

**Comment by RobTillaart** [comment]
@oduled 
Is this still an issue as the output is explained ?
If not, please close the issue.
